File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n=len(height)
        left=0
        right=n-1
        maxArea=min(height[left],height[right])*(right-left)
        while(left<right):
            area=min(height[left],height[right])*(right-left)
            if (area>maxArea):
                maxArea=area
            if(height[left]<height[right]):
               left=left+1
            else:
               right=right-1
        return maxArea

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n=height.size();
        int left=0;
        int right=n-1;
        int area;
        int maxArea=min(height[left],height[right])*(right-left);
        while(left<right){
        area=min(height[left],height[right])*(right-left);
        if(area>maxArea){
            maxArea=area;
        }
        if(height[left]<height[right]){
        left++;
        }
        else{
        right--;
        }
        }
        return maxArea;
    }
};